tpm2/structures.go

```diff
@@ -1021,10 +1021,37 @@ type AuthCommand struct {
 	Auth       tpmutil.U16Bytes
 }
 
+// BDigest implements the structure of TPM2B_Digest as described in Part 2 Structures, Page 97
+type BDigest struct {
+	Size   uint16
```

Comment:
What does this type add that is missing from U16Bytes? For example, Size here is len(Digest).

Reply:
Yep. I agree, but spec disagrees.

Reply:
The spec wire format is `size uint16 || data []byte`, but that doesn't mean that the in-memory representation in go-tpm should have a separate size field [^1] - part of the job of go-tpm is to translate to and from a higher level representation (like []byte) to a lower level representation (like a correctly serialized TPM2B_DIGEST).

Indeed the entire purpose of tpmutil.U16Bytes is to allow Go code to treat as a boring byte slice something that needs to be serialized into the TPM2B format. So I maintain that the proper fix to the serialization errors you're seeing here is not to create a new type for TPM2B_DIGEST [^2], but to figure out why the full TPM2_PolicyOR is not serializing as expected [^3].

[^1]: If we have to track a separate size field that is set independently of the data, now we need consistency checks in **many** places
[^2]: I'd actually be totally shocked if this fix was the first time go-tpm has needed to change up the way it represents a TPM2B_DIGEST. Those are everywhere.
[^3]: I think TPMLDigest just needs an override `Encode`/`Decode...` pair because the automatic one just isn't doing the trick.

Reply:
I found a way to handle that. Thank you :)